starlight-rabbitmq/src/main/java/com/datastax/oss/starlight/rabbitmq/AMQChannel.java

```diff
@@ -1673,6 +1672,14 @@ private void deliverCurrentMessageIfComplete() {
                         exchangeName,
                         throwable);
                   }
+
+                } else if (throwable.getCause()
+                    instanceof PulsarClientException.AlreadyClosedException) {
+                  LOGGER.error(
+                      "Failed to send message to exchange='{}' because producer is closed. Removing producer from cache",
+                      exchangeName,
+                      throwable);
+                  getGatewayService().removeProducer(producer.getTopic());
```

Comment:
We should remove the Producer only if the cached instance is this instance.
Removing by topic name may lead to races